src/bencode_parser.hpp

```diff
@@ -0,0 +1,19 @@
+//
+// Created by nicof on 28/07/22.
+//
+
+#pragma once
+
+#include <string>
+
+#include "bencode_value.hpp"
+
+namespace bencode {
+  class BencodeParser {
+   public:
+    BencodeParser() = default;
+    ~BencodeParser() = default;
+    BencodeValue decode(std::string &decoded);
```

Comment:
I think the return value needs to be a `unique_ptr` otherwise the extra fields from the subclasses are truncated.